### PR TITLE
Run Conjur cluster as a stateful set

### PR DESCRIPTION
**Pull request by dustinmm80**
This PR modifies Conjur master and standbys as a StatefulSet with persistent volumes in Kubernetes 1.5+ and OpenShift 3.5+. This makes it possible to restore the Conjur master manually when its pod is rescheduled.

Changes:
- Runs the Conjur master and standbys as a stateful set, with persistent volumes.
- A relaunch_master script is now available that terminates the master and reconfigures it. This script runs in Jenkins pipeline.
- Configures Conjur standbys and followers in parallel.
- haproxy now runs more properly, with default entrypoint and configuration reloading.

See Jenkins builds here: https://jenkins.conjur.net/view/k8s/job/cyberark--kubernetes-conjur-deploy/job/statefulset/

This PR replaces PR #7 .

To verify:
- Clone this branch
- Run demo scripts against GKE and OpenShift 3.7 and verify they work

### CODE REVIEW

**Comment by dustinmm80** [comment]
We don't have persistent volumes set up for our 3.7 cluster yet, adding that setup.

```sh-session
$ oc describe pvc conjur-data-conjur-cluster-0
Name:		conjur-data-conjur-cluster-0
Namespace:	conjur-dustinc
StorageClass:
Status:		Pending
Volume:
Labels:		app=conjur-node
Annotations:	<none>
Capacity:
Access Modes:
Events:
  FirstSeen	LastSeen	Count	From				SubObjectPath	Type		Reason		Message
  ---------	--------	-----	----				-------------	--------	------		-------
  9m		3m		26	persistentvolume-controller			Normal		FailedBinding	no persistent volumes available for this claim and no storage class is set

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,8 @@
 .DS_Store
 standby-seed.tar
 haproxy.cfg
+!build/haproxy/haproxy.cfg
 tmp/
 output/
+notes.txt
+setenv.sh
```
Comment (jtuttle) [inline .gitignore]: Did you want to leave this gitignore stuff in to ease your development experience? If not we should probably remove it.

File: 6_configure_standbys.sh [hunk 2]
```diff
@@ -29,8 +31,14 @@ for pod_name in $pod_list; do
 
   $cli exec $pod_name -- evoke unpack seed /tmp/standby-seed.tar
   $cli exec $pod_name -- evoke configure standby -i $master_pod_ip
+}
+
+for pod_name in $pod_list; do
+  configure_standby $pod_name &
```
Comment (jtuttle) [inline 6_configure_standbys.sh]: Oh cool, you parallelized the standby setup? Nice!

File: .gitignore (hunk 1 above)
Comment (jtuttle) [reply .gitignore]: Dustin would prefer to keep this, so leaving it in.

**Pull request merged**